common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationAwareUTF8String.java

```diff
@@ -1434,6 +1435,42 @@ public static UTF8String[] icuSplitSQL(final UTF8String string, final UTF8String
     return strings.toArray(new UTF8String[0]);
   }
 
+  /**
+   * Splits the `string` into an array of substrings based on the `delimiter` regex, with respect
+   * to the maximum number of substrings `limit`.
+   *
+   * @param string the string to be split
+   * @param delimiter the delimiter regex to split the string
+   * @param limit the maximum number of substrings to return
+   * @return an array of substrings
+   */
+  public static UTF8String[] split(final UTF8String string, final UTF8String delimiter,
+      final int limit, final int collationId) {
+    CollationFactory.Collation collation = CollationFactory.fetchCollation(collationId);
+    assert collation.isUtf8BinaryType || collation.isUtf8LcaseType :
+        "Unsupported collation type for split operation.";
+
+    if (CollationFactory.fetchCollation(collationId).isUtf8BinaryType) {
```

Comment:
no need to call `CollationFactory.fetchCollation` again here, we already have the collation